### DISCUSSION TITLE
How to validate EC2::InstanceType?

### DISCUSSION THREAD

**Discussion by jkmartin19** [aws/aws-sdk-cpp#2444] [Q&A] [answered]
Hello,

I need to validate that a string is a valid EC2 instance type. I was hoping code like this would work:

```
type = Aws::EC2::Model::InstanceTypeMapper::GetInstanceTypeForName(some_string);
if (type == Aws::EC2::Model::InstanceType::NOT_SET) {
    // handle invalid instance type
} else {
   // assume valid instance type
}
```

The issue is that the above code will return a seemingly valid InstanceType because GetInstanceTypeForName() uses the EnumParseOverflowContainer, so it will return an InstanceType that is something other than NOT_SET. 

I'm wondering what the correct way to validate that a string input is a valid InstanceType. Any pointers would be very appreciated.

**Comment by jkmartin19** [comment]
Is there anyone who knows how to validate that a string is a valid instance type? I'm hoping I don't have to write something myself.

**Reply by jmklix** [reply, marked as answer]
You can use [DescribeInstanceTypeRequest/Response](https://sdk.amazonaws.com/cpp/api/LATEST/aws-cpp-sdk-ec2/html/class_aws_1_1_e_c2_1_1_model_1_1_describe_instance_types_request.html) and compare that to your string to see if that is a valid instance. You can look [here](https://docs.aws.amazon.com/code-library/latest/ug/ec2_example_ec2_DescribeInstanceTypes_section.html) for examples of how to use that in other languages (sorry we don't currently have an example in c++)
